### PR TITLE
Clean up the connection generation and handling in blaze client

### PR DESCRIPTION
**Pull request by bryce-anderson**
__All the changes here are purely cosmetic in nature.__

Really I'm just beating around the bush of the big problem: `Http1ClientStage`.
### Changes:
* Removed the `ConnectionBuilder` type because shutdown is not needed and other than that its just a function.
* `Http1Support` is no longer such a Frankenstein. The implementation is now private and the
object has a constructor that returns a `ConnectionBuilder` function.
* Make The `ConnectionManager` implementations private and generated through construtor functions much like
the new `Http1Support`.
* Moved private stuff out of the `package` object and into the private `bits` object.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
cc #402

File: blaze-client/src/main/scala/org/http4s/client/blaze/ConnectionManager.scala [hunk 1]
```diff
@@ -4,6 +4,13 @@ import org.http4s.Request
 
 import scalaz.concurrent.Task
 
+/** type that is responsible for the client lifecycle
+  *
+  * The [[ConnectionManager]] is a general wrapper around a [[ConnectionBuilder]]
+  * that can pool resources in order to conserve on resources such as socket connections,
```
Comment (rossabaker) [inline blaze-client/src/main/scala/org/http4s/client/blaze/ConnectionManager.scala]: "conserve resources"

File: blaze-client/src/main/scala/org/http4s/client/blaze/BasicManager.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package org.http4s
+package client
+package blaze
+
+import scalaz.concurrent.Task
+
+
+/** Create a basic [[ConnectionManager]] that creates new connections on each request */
+object BasicManager {
+  def apply(builder: ConnectionBuilder): ConnectionManager =
```
Comment (rossabaker) [inline blaze-client/src/main/scala/org/http4s/client/blaze/BasicManager.scala]: Maybe `ConnectionManager.basic` and `ConnectionManager.pooled`?  I wouldn't necessarily know where to look for the scattered connection managers.

**Comment by rossabaker** [comment]
Does this `bits` object have a precedent somewhere else?  I don't mind it, but it's not something I've seen outside of our code base.

File: blaze-client/src/main/scala/org/http4s/client/blaze/BasicManager.scala (hunk 2 above)
Comment (bryce-anderson) [inline blaze-client/src/main/scala/org/http4s/client/blaze/BasicManager.scala]: Good idea.

**Comment by bryce-anderson** [comment]
I can't remember where I've seen it called `bits` before, but I believe there are a few projects that use `utils` or other named things. In spirit I see it like the haskell `Internal` packages, though I arbitrarily name it bits.

**Comment by rossabaker** [comment]
:+1: when Travis approves.

**Pull request merged**